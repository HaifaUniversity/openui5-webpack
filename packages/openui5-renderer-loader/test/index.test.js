import loader from '../src/index';

describe('Renderer Loader', () => {
  it('should handle control filter', (done) => {
    const content = 'test-content';
    const addDependency = jest.fn();
    loader.apply({
      query: {
        filterRegEx: /[/\\]src[/\\](.*)\.js$/
      },
      async: () => (err, data) => {
        expect(addDependency.mock.calls.length).toBe(1);
        expect(addDependency.mock.calls[0][0]).toEqual('abc/TestRenderer');
        expect(data).toMatchSnapshot();
        done();
      },
      resolve: (context, name, callback) => {
        callback();
      },
      addDependency,
      resourcePath: 'test/src/abc/Test.js',
    }, [content]);
  })
});
