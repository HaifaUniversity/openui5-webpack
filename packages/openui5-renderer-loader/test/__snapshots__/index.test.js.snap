// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer Loader should handle control filter 1`] = `
"test-content
sap.ui.define(['jquery.sap.global'], function(jQuery) {
jQuery.sap.setObject(\\"abc.TestRenderer\\", require(\\"abc/TestRenderer\\"));
});"
`;
